### ISSUE TITLE
Multiple regressions for components inside dialog in flow 23.3.29

### ISSUE THREAD

**Issue by davidef** [vaadin/flow-components#5818]
### Description

The fixed reported in the 23.3.29 release notes:
Focus the closest focusable on overlay mousedown (https://github.com/vaadin/web-components/pull/6850) (https://github.com/vaadin/web-components/pull/6856)
cause multiple regressions for components inside a dialog.

### Expected outcome

Work as it was in 23.3.28

### Minimal reproducible example

```
	public TestView() {
		add(new Button("Open DragAndDrop", e -> {
			Dialog dialog = new Dialog();
			dialog.add(new MyLayout());
			dialog.open();
		}));
		
		add(new MyLayout());
		
		add(new Button("Open TextField", e -> {
			Dialog dialog = new Dialog();
			TextField textField = new TextField();
			textField.setValue("abc");
			dialog.add(textField);
			dialog.open();
		}));
		
		TextField textField = new TextField();
		textField.setValue("abc");
		add(textField);
	}
	
	public static class MyLayout extends VerticalLayout{
		List<String> values = new ArrayList<>(List.of("A","B","C"));
		
		public MyLayout() {			
			for (final String value:values){
				Button valueButton = new Button(value);
				
				DragSource<Button> valueButtonDragSource = DragSource.create(valueButton);
				valueButtonDragSource.setDragData(value);
				valueButtonDragSource.setDraggable(true);
				
				DropTarget<Button> valueButtonDropTarget = DropTarget.create(valueButton);
				valueButtonDropTarget.setDropEffect(DropEffect.MOVE);
				valueButtonDropTarget.addDropListener(event -> {
				    if (event.getDropEffect() == DropEffect.MOVE) {
				    	String dropTargetData = (String) event.getDragData().get();
				    	
				    	Button dragSource = (Button) event.getDragSourceComponent().orElse(null);
				    	Button dropTarget = event.getSource();

				    	int sourceIndex = getElement().indexOfChild(dragSource.getElement());
				    	int dropIndex = getElement().indexOfChild(dropTarget.getElement());
						
				    	if (sourceIndex >= 0 && dropIndex >= 0 && sourceIndex != dropIndex) {
				    		String dropValue = values.remove(sourceIndex);
				    		remove(dragSource);
				    		if (dropValue != dropTargetData) {
				    			throw new IllegalArgumentException("Unexpected data");
				    		}
				    		if (dropIndex > sourceIndex) {
				    			values.add(dropIndex, dropValue);
				    			getElement().insertChild(dropIndex, dragSource.getElement());
				    		}else {
				    			values.add(dropIndex, dropValue);
				    			getElement().insertChild(dropIndex, dragSource.getElement());
				    		}
				    	}
				    }
				});
					
				add(valueButton);
			}
		}
	}
```

### Steps to reproduce

In the example:
1. drag and drop works fine outside the dialog but not when in the dialog
2. double click on the text in the TextField to select it. It works fine in the main view but not when in the dialog.

### Environment

Vaadin version(s): 23.3.29

**Comment by marcushellberg** [comment]
Sounds like this might be related to https://github.com/vaadin/flow-components/issues/5507

**Comment by knoobie** [comment]
It's related. The root cause is already fixed and release soon. See: https://github.com/vaadin/flow/issues/18221#issuecomment-1840929113

**Comment by web-padawan** [comment]
> Focus the closest focusable on overlay mousedown (https://github.com/vaadin/web-components/pull/6850) (https://github.com/vaadin/web-components/pull/6856)
cause multiple regressions for components inside a dialog.

The PR has been reverted and released in 23.3.30: https://github.com/vaadin/web-components/pull/6886#issuecomment-1855545903

**Comment by davidef** [comment]
@web-padawan we just tested 23.3.30 and the issue is now fixed. Thank you
